packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart

```diff
@@ -112,13 +115,29 @@ class MapConfiguration {
   ///
   /// See https://developers.google.com/maps/documentation/get-map-id
   /// for more details.
-  final String? cloudMapId;
+  final String? mapId;
 
   /// Locally configured JSON style.
   ///
   /// To clear a previously set style, set this to an empty string.
   final String? style;
 
+  /// The type of marker that the map should use.
+  ///
+  /// Advanced and legacy markers could be handled differently by platform
+  /// implementations. This property indicates which type of marker should be
+  /// used.
+  final MarkerType? markerType;
```

Comment:
What does a null type indicate? If that's a distinct meaningful state, the doc comment needs to explain it. (If it's not, it shouldn't be nullable.)

Reply:
It only indicates that marker type wasn't set explicitly. It would be good to make it non-nullable but there could be complications with `diffFrom`, `applyDiff` and `isEmpty` (null is used to check if value is set or empty). How should it look there? If `isEmpty` ignores `markerType` value because it's always set it could break `isEmpty` check and result in missing or excessive updates (for example, controller is not updated with new map configuration if options are empty in `google_map.dart`)

Reply:
Sorry about that, with just the local context of the change I forgot how this object works:

```dart
  /// Any options that aren't passed will be null, which allows this to serve
  /// as either a full configuration selection, or an update to an existing
  /// configuration where only non-null values are updated.
```

So for the unusual case of this object, nullable is indeed correct.